contracts/market/template/MarketCore.sol

```diff
@@ -35,30 +35,27 @@ contract MarketCore is AccessControlledAndUpgradeableModifiers, IMarketCommon, I
   /// @notice This calculates the value transfer from the overbalanced to underbalanced side (i.e. the funding rate)
   /// This is a further incentive measure to balanced markets. This may be present on some and not other pool token markets.
   /// @param overbalancedIndex poolType with more liquidity.
-  /// @param overbalancedValue Side with more liquidity.
-  /// @param underbalancedValue Side with less liquidity.
+  /// @param overbalancedEffectiveValue Side with more liquidity.
+  /// @param underbalancedEffectiveValue Side with less liquidity.
   /// @return fundingAmount The amount the overbalanced side needs to pay the underbalanced.
   function _calculateFundingAmount(
     uint256 overbalancedIndex,
-    uint256 overbalancedValue,
-    uint256 underbalancedValue
+    uint256 overbalancedEffectiveValue,
+    int256 floatLeverage,
+    uint256 underbalancedEffectiveValue
   ) internal view returns (int256[2] memory fundingAmount) {
-    /*
-    totalFunding is calculated on the notional of between long and short liquidity and 2x long and short liquidity. 
-    The notional on which funding is calculated increaseas as the imbalance increases.
-    The split of funding paid between underbalanced and overbalanced is 50/50 when long = short liq and moves
-    linearly to 100% paid by the overbalanced beyong the point where liquidity is 2:1 ratio
-    (liquidity in overbalanced side is more than double liquidity in underbalanced side)
-    This modular function is logical but naive implementation that will likely change somewhat upon more indepth 
-    modelling results that are still pending. 
-    */
+    uint256 totalUserEffectiveValue = overbalancedEffectiveValue + underbalancedEffectiveValue;
 
-    // fundingRateMultiplier is in basis points so need to divide by 10,000.
-    uint256 totalFunding = (2 * overbalancedValue * fundingRateMultiplier * oracleManager.EPOCH_LENGTH()) / (365.25 days * 10000);
+    FundingVariables memory fundingVariablesMem = fundingVariables;
+    // NOTE: fundingRateMultiplier is in basis points so need to divide by 10,000, and floatLeverage is in base 1e18 so overall we divide by 1e22.
+    uint256 totalFunding = (totalUserEffectiveValue *
+      fundingVariablesMem.fundingRateMultiplier *
+      Math.max(fundingVariablesMem.minFloatPoolFundingBoost, floatLeverage.abs()) *
+      oracleManager.EPOCH_LENGTH()) / (365.25 days * 1e22);
 
     uint256 overbalancedFunding = Math.min(
       totalFunding,
-      (totalFunding * ((2 * overbalancedValue) - underbalancedValue)) / (overbalancedValue + underbalancedValue)
+      (totalFunding * ((2 * overbalancedEffectiveValue) - underbalancedEffectiveValue)) / (totalUserEffectiveValue)
```

Comment:
No change here, just using the `totalUserEffectiveValue` variables instead of `(overbalancedValue + underbalancedValue)`